### PR TITLE
feat: add method to get the current OAuth token from client

### PR DESCRIPTION
**Pull request by elivlo**
Hi, this should resolve Issue #169.
@strideynet

### CODE REVIEW

**Comment by strideynet** [comment]
Hey! Thanks for raising this PR. Do you think you'd be able to include a test with this ? If you are unsure, please reach out to me on Slack and I can give you a hand.

File: spotify.go [hunk 1]
```diff
@@ -307,3 +309,16 @@ func (c *Client) NewReleases(ctx context.Context, opts ...RequestOption) (albums
 
 	return &result, nil
 }
+
+// Token gets the client's current token.
+func (c *Client) Token() (*oauth2.Token, error) {
+	transport, ok := c.http.Transport.(*oauth2.Transport)
+	if !ok {
+		return nil, errors.New("spotify: oauth2 transport type not correct")
```
Comment (strideynet) [inline spotify.go:317]: Perhaps "spotify: client not backed by oauth2 transport"

**Comment by elivlo** [comment]
> Hey! Thanks for raising this PR. Do you think you'd be able to include a test with this ? If you are unsure, please reach out to me on Slack and I can give you a hand.

Sure I will do, but unfortunately the link you posted in #168 is not valid anymore.

**Comment by strideynet** [comment]
Does this link work http://invite.slack.golangbridge.org/ ?

**Comment by strideynet** [comment]
https://join.slack.com/t/gophers/shared_invite/zt-urapn2hs-qPtfq~6PQzTYJtysXmG3jg <- that might work for now

File: spotify_test.go [hunk 2]
```diff
@@ -101,3 +103,58 @@ func TestNewReleasesRateLimitExceeded(t *testing.T) {
...
+		t.Errorf("Invalid access token data: %s", token.AccessToken)
+	}
+	if token.RefreshToken != "refresh_token" {
+		t.Errorf("Invalid refresh token data: %s", token.RefreshToken)
+	}
+	if token.TokenType != "test_type" {
+		t.Errorf("Invalid token type: %s", token.TokenType)
+	}
+
+	// create and test client without an oauth token
```
Comment (strideynet) [inline spotify_test.go:144]: Where possible, break seperate cases out into subtests (https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks)

So in this case, you'd have a subtest for "success", "non oauth2 transport" and "invalid token"

It makes it a bit clearer what each case in the test is :)

Comment (elivlo) [reply spotify_test.go:144]: Yep. This is a nice way to test a function for all test cases I haven't heard of it yet.
I hope I fixed it correctly.

**Comment by elivlo** [comment]
> https://join.slack.com/t/gophers/shared_invite/zt-urapn2hs-qPtfq~6PQzTYJtysXmG3jg <- that might work for now

Thanks, it worked!

**Pull request merged**